lib/Config.js

```diff
@@ -906,6 +906,20 @@ class Config extends EventEmitter {
             assert(typeof config.replicationGroupId === 'string',
                 'bad config: replicationGroupId must be a string');
             this.replicationGroupId = config.replicationGroupId;
+        } else if (process.env.HOSTNAME) {
+            // Extract the hash part from the pod name (e.g., cloudserver-7869f99955-nr984)
+            const parts = process.env.HOSTNAME.split('-');
+            // Kubernetes pods typically have two UID parts:
+            // - A long hash (10 characters) that guarantees uniqueness within the namespace.
+            // - A short hash (5 characters) that is unique only within the current ReplicaSet/Deployment.
+            if (parts.length > 2) {
+                const longId = parts[parts.length - 2];
+                const shortId = parts[parts.length - 1];
+                this.replicationGroupId = longId.slice(0, 2) + shortId;
+            } else {
+                // fallback: just use the last part if format is unexpected
+                this.replicationGroupId = parts.pop();
+            }
         } else {
             this.replicationGroupId = 'RG001';
```

Comment:
We may one day hit the same issue if we have a "CRR" configuration on Artesca that uses a different replication group ID, right? Could be worth ensuring that if we're in an Artesca setting (how?), we don't have a replication group ID configured, and maybe refuse to start if that's the case.